sdk/storage/src/data_lake/clients/file_system_client.rs

```diff
@@ -6,41 +6,29 @@ use azure_core::pipeline::Pipeline;
 use azure_core::{Context, HttpClient, PipelineContext};
 use bytes::Bytes;
 
-use std::sync::Arc;
 use url::Url;
 
-pub trait AsFileSystemClient<A: Into<String>> {
-    fn as_file_system_client(&self, name: A) -> Result<Arc<FileSystemClient>, url::ParseError>;
-}
-
-impl<A: Into<String>> AsFileSystemClient<A> for Arc<DataLakeClient> {
-    fn as_file_system_client(&self, name: A) -> Result<Arc<FileSystemClient>, url::ParseError> {
-        FileSystemClient::new(self.clone(), name.into())
-    }
-}
-
 #[derive(Debug, Clone)]
 pub struct FileSystemClient {
-    data_lake_client: Arc<DataLakeClient>,
+    data_lake_client: DataLakeClient,
     name: String,
     url: Url,
 }
 
 impl FileSystemClient {
-    pub(crate) fn new(
-        data_lake_client: Arc<DataLakeClient>,
-        name: String,
-    ) -> Result<Arc<Self>, url::ParseError> {
-        let mut url = data_lake_client.url().to_owned();
+    pub(crate) fn new(data_lake_client: DataLakeClient, name: String) -> Self {
+        let mut url = url::Url::parse(data_lake_client.url().as_str()).unwrap();
```

Comment:
Panicking here to avoid producing a ParseError. Could be replaced by the approach CosmosDB is using, which is not parsing the URL and just using the strings which will result in HTTP errors if they are wrong.